test/text_tags_test.js

```diff
@@ -400,4 +400,31 @@ describe('TextTags', function() {
       assert.equal("My Title", seo.getTitle(config, 'home-page', {data: {title: "My Title"}}, {}));
     });
   });
+
+  describe('Tag Page', function() {
+    it("Generates SEO tags for a tag page", function () {
+      const seoConfig = {
+        generators: [TextTags],
+      }
+      const tag = {slug: "Foobar", description: "Some Foobar", name: "Footag", path: "/Foopath"}
+      const string = getSeoMetadata(seoConfig, {"sketches-host": "http://foo.com"}, 'tag-page', {data: {tag: tag}}, {url: url.parse("/my-page")})
+      assertContains('<title>Footag</title>', string);
+      assertContains('<meta name="description" content="Some Foobar"/>', string);
+      assertContains('<meta name="keywords" content="Footag"/>', string);
+      assertContains('<link rel="canonical" href="http://foo.com/Foopath"/>', string);
+    });
+    it("gets tag data as fallback if the tag metadata is falsy", function () {
+      const seoConfig = {
+        generators: [TextTags],
+      }
+      const tag = {}
+      const config = {"seo-metadata": [{"owner-type": "home", "data": {'page-title': "Foobar", 'description': "Some Foobar", 'keywords': "keywords", 'canonicalUrl': "http://foo.com"}}]};
+      const string = getSeoMetadata(seoConfig, config, 'home-page', {}, {url: url.parse("/")});
```

Comment:
you should be asking for tag-page, and since tag data was empty it fallbacks for the home-page data internally.